client/allocrunner/taskrunner/task_runner_hooks.go

```diff
@@ -235,9 +235,11 @@ func (tr *TaskRunner) prestart() error {
 		}
 
 		// Run the prestart hook
-		// use a joint context to allow any blocking pre-start hooks
+		// use a join context to allow any blocking pre-start hooks
 		// to be canceled by either killCtx or shutdownCtx
-		joinedCtx, _ := joincontext.Join(tr.killCtx, tr.shutdownCtx)
+		joinedCtx, joinedCancel := joincontext.Join(tr.killCtx, tr.shutdownCtx)
+		defer joinedCancel()
```

Comment:
Is a new `joinedCtx` per hook necessary? We can reuse the same one for all prestart hook. 

Also, I haven't touched this code in so long - would this accidentally risk stopping goroutines that launched by prestart hooks if they rely on the passed context? If we use a single joinedCtx, we can save it and cancel it in exited/stop hooks.

Reply:
> Is a new joinedCtx per hook necessary? We can reuse the same one for all prestart hooks.

Oh, that's a good point. We could move this up to the top of the loop (but we can't pass it into this `prestart` method from the caller in `task_runner.go` because then we don't get to use the `defer`). Will do that.

>  would this accidentally risk stopping goroutines that launched by prestart hooks if they rely on the passed context?

Yes, but I went through and verified we're not doing that currently (as @lgfa29 has noted most of the prestart hooks don't even use the context). I feel _fairly_ confident saying that saving the context in a goroutine in a prestart hook is not a behavior we should be doing at all, given how we treat prestart hooks? I can add a note to warn future developers about it, at least.